ui/pages/groups/index.js

```diff
@@ -115,111 +115,117 @@ function Details({ group, admin, onDelete }) {
-                  method: 'PATCH',
-                  body: JSON.stringify({ usersToAdd }),
-                })
+  return (
+    !loading && (
+      <div className='flex flex-1 flex-col space-y-6'>
+        {admin && (
+          <>
+            <section>
+              <h3 className='mb-4 border-b border-gray-800 py-4 text-3xs uppercase text-gray-400'>
+                Access
+              </h3>
+              <GrantsList
+                grants={grants}
+                onRemove={async id => {
+                  await fetch(`/api/grants/${id}`, { method: 'DELETE' })
+                  mutateGrants({ items: grants.filter(x => x.id !== id) })
+                }}
+                onChange={async (privilege, grant) => {
+                  const res = await fetch('/api/grants', {
+                    method: 'POST',
+                    body: JSON.stringify({
+                      ...grant,
+                      privilege,
+                    }),
+                  })
 
-                await mutateUsers(`/api/users?group=${group.id}`)
-                setEmails([])
-              }}
-            />
-            <div className='mt-4'>
-              {users?.length === 0 && (
+                  // delete old grant
+                  await fetch(`/api/grants/${grant.id}`, { method: 'DELETE' })
+                  mutateGrants({
+                    items: [
+                      ...grants.filter(f => f.id !== grant.id),
+                      await res.json(),
+                    ],
+                  })
+                }}
+              />
+              {!grants?.length && (
                 <EmptyData>
-                  <div className='mt-6'>No members in the group</div>
+                  <div className='mt-6'>No access</div>
                 </EmptyData>
               )}
-              <IdentityList
-                list={users}
-                onClick={async userId => {
-                  const usersToRemove = [userId]
+            </section>
+            <section>
+              <h3 className='mb-2 border-b border-gray-800 py-4 text-3xs uppercase text-gray-400'>
+                Users{users?.length > 0 && <span> ({users.length})</span>}
+              </h3>
+              <EmailsSelectInput
+                selectedEmails={emails}
+                setSelectedEmails={setEmails}
+                existMembers={existMembers}
+                onClick={async () => {
+                  const usersToAdd = emails.map(email => email.id)
+                  console.log(emails)
                   await fetch(`/api/groups/${id}/users`, {
                     method: 'PATCH',
-                    body: JSON.stringify({ usersToRemove }),
-                  })
-                  mutateUsers({
-                    items: users.filter(i => i.id !== userId),
+                    body: JSON.stringify({ usersToAdd }),
                   })
+
+                  mutateUsers({ items: [...users, ...emails] })
```

Comment:
instead of calling the endpoint like this `mutateUsers(`/api/users?group=${group.id}`)`, we can just do `items: [...users, ...emails]` to update the  dom